pilot/pkg/networking/core/envoyfilter/rc_patch.go

```diff
@@ -94,38 +91,38 @@ func patchVirtualHosts(patchContext networking.EnvoyFilter_PatchContext,
 			IncrementEnvoyFilterMetric(rp.Key(), VirtualHost, false)
 		}
 	}
-	if removedVirtualHosts.Len() > 0 {
-		routeConfiguration.VirtualHosts = slices.FilterInPlace(routeConfiguration.VirtualHosts, func(virtualHost *route.VirtualHost) bool {
-			return !removedVirtualHosts.Contains(virtualHost.Name)
-		})
-	}
+
+	routeConfiguration.VirtualHosts = slices.FilterInPlace(routeConfiguration.VirtualHosts, func(virtualHost *route.VirtualHost) bool {
```

Comment:
Technically speaking isn't this a behavior change? Before we could "remove" a Vhost we also Added.

However I think this is probably better

Reply:
**old behavior**: delete first, then add, because it is already in `removedVirtualHosts`, it is finally deleted
**new behavior**: delete first, then add, because there is no `removedVirtualHosts` control, it is finally added

This is indeed a behavior change, do we need to keep it ? If so, i can revert this change.

Reply:
I think the old behavior looks like a bug 😀.

Reply:
I remember not only vhost， every other poatch types performs remove at last

Reply:
I just see you changed that in the previous pr, I would suggest revert this change

Reply:
> I remember not only vhost， every other patch types performs remove at last

The difference between there and other places is:
Here it is explicitly specified which CRs to delete, while other places use "empty or nil" to check.

Reply:
> I would suggest revert this change

only revert this PR's this place?

Reply:
@ramaraochavali What do you think?

Reply:
Keep all previous remove at last, or it is a bad incompatability. At some point It violates older config takes priority

Reply:
Sorry was off friday. So could not respond. What is the previous PR? removes at last was the choice we made IIRC

Reply:
this previous is #51791 (LDS related cleanup), this PR is RDS related.

Reply:
thanks all~ reverted

Reply:
Do you revert previous pr?

Reply:
revert #51791 ? No, i did't, why do we need to revert #51791 ?

Reply:
If if changed remove logic, you should revert that